src/Query/SelectQuery.php

```diff
@@ -185,15 +186,16 @@ protected function applySelect(AsyncMysqlConnection $conn, dataset $data): datas
               }
             } else {
               $col = C\last($parts);
-              if (!C\contains_key($formatted_row, $col)) {
-                $formatted_row[$col] = $val;
+              if ($col is nonnull) {
+                $formatted_row[$col] ??= $val;
```

Comment:
This expression should also check for keys. However, it is unclear to me what HHVM4.0 would do if you did try to assign to `$dict[null]`. HHVM4.15 throws an InvalidArgumentException. This nonnull would not be needed if `C\lastx` would be safe.